modules/dynamic-plugins/proc-export-third-party-plugins-rhdh.adoc

```diff
@@ -0,0 +1,116 @@
+
+* *Embedded dependencies* are bundled into the dynamic plugin package with their dependencies hoisted to the top level. By default, packages with `-node` or `-common` suffixes are embedded.
++
+You can use the `--embed-package` flag to specify additional embedded packages, including those in the same repository that do not follow the default naming convention.
+
+The following is an example of exporting a dynamic plugin with shared and embedded packages:
+
+.Example dynamic plugin export with shared and embedded packages
+[source,bash]
+----
+npx @janus-idp/cli@latest export-dynamic-plugin --shared-package '!/@backstage/plugin-notifications/'<1> --embed-package @backstage/plugin-notifications-backend <2>
+----
+
+<1> `@backstage/plugin-notifications` package is treated as a private dependency and is bundled in the dynamic plugin package, despite being in the `@backstage` scope.
+<2> `@backstage/plugin-notifications-backend` package is marked as an embedded dependency and is bundled in the dynamic plugin package.
+--
+
+Front-end plugins::
++
+--
+Front-end plugins can use `scalprum` for configuration, which the CLI can generate automatically during the export process. The generated default configuration is logged when running the following command:
+
+.Example command to log the default configuration
+[source,bash]
+----
+npx @janus-idp/cli@latest export-dynamic
+----
+
+The following is an example of default `scalprum` configuration:
+
+.Default `scalprum` configuration
+[source,json]
+----
+"scalprum": {
+  "name": "<package_name>",  // The Webpack container name matches the NPM package name, with "@" replaced by "." and "/" removed.
+  "exposedModules": {
+    "PluginRoot": "./src/index.ts"  // The default module name is "PluginRoot" and doesn't need explicit specification in the app-config.yaml file.
+  }
+}
+----
+
+You can add a `scalprum` section to the `package.json` file. For example:
+
+.Example `scalprum` customization
+[source,json]
+----
+"scalprum": {
+  "name": "custom-package-name",
+  "exposedModules": {
+    "FooModuleName": "./src/foo.ts",
+    "BarModuleName": "./src/bar.ts"
+    // Define multiple modules here, with each exposed as a separate entry point in the Webpack container.
+  }
+}
+----
+
+Dynamic plugins might need adjustments for {product-short} needs, such as static JSX for mountpoints or dynamic routes. These changes are optional but might be incompatible with static plugins.
+
+To include static JSX, define an additional export and use it as the dynamic plugin's `importName`. For example:
+
+.Example static and dynamic plugin export
+[source,tsx]
+----
+// For a static plugin
+export const EntityTechdocsContent = () => {...}
+
+// For a dynamic plugin
+export const DynamicEntityTechdocsContent = {
+  element: EntityTechdocsContent,
+  staticJSXContent: (
+    <TechDocsAddons>
+      <ReportIssue />
+    </TechDocsAddons>
+  ),
+};
+----
+--
+
+.Prerequisites
```

Comment:
This (together with Procedure) should be at the begging of this document, right after the first introductory paragraph.  This is common for both frontend and backend

Reply:
The backend and frontend plugins specify information about the dependencies, which are also listed as prerequisites. I suggest we present the conceptual information about the dependencies first, followed by the prerequisites. This structure would make the procedure clearer for users as they follow the steps.
WDYT?

Reply:
Consider splitting in multiple procedures:

*   _Exporting a third-party backend plugin_
*    _Exporting a third-party front-end plugin_

Consider also moving this Prequisistes section at the top of each procedure.

Then, consider clarifying lines 8 to 99 to make clear if these are very long prerequisites lists, or a first step "Prepare your plugin for exportation". (Seems more like a first step)

Reply:
The current structure doesn't make sense. :-/ 
Everything below this line is common for both frontend and backend plugins and as such, I think it is better if it is at the beginning of this document as this will be enough for 90% of the use cases. After this generic explanation, we go deeper into the specifics of frontend and backend plugins.